d2graph/layout.go

```diff
@@ -41,6 +41,7 @@ func (g *Graph) ExtractAsNestedGraph(obj *Object) *Graph {
 	descendantObjects, edges := pluckObjAndEdges(g, obj)
 
 	tempGraph := NewGraph()
+	tempGraph.RootLevel = int(obj.Level()) - 1
```

Comment:
is it possible for this to go negative?

Reply:
It should not be possible. For example with `a.b.c`, with the least nested `a` being extracted, its Level is 1 (1 + 0 for root), and then the nested RootLevel is stored as 0. For `b` its Level is 2 so the RootLevel would be 1. Even if `a` were extracted twice for some reason, its Level is still 1 in the nested graph, so the root level would remain 0.